tests/std/tests/Dev11_0836436_get_time/test.cpp

```diff
@@ -905,3 +908,24 @@ void test_gh_2848() {
         assert(err == (ios_base::eofbit | ios_base::failbit));
     }
 }
+
+void test_gh_4820() {
+    // GH-4820 <iomanip>: std::put_time should copy unknown conversion specifiers instead of crash
+    time_t t = time(nullptr);
+    tm currentTime;
+    localtime_s(&currentTime, &t);
+
+    {
+        wstringstream wss;
+        wss << put_time(&currentTime, L"%Ei%!%E%J%P");
+        assert(wss.rdstate() == ios_base::goodbit);
+        assert(wss.str() == L"%Ei%!%E%J%P");
+    }
+
+    {
+        stringstream ss;
+        ss << put_time(&currentTime, "%Ei%!%E%J%P");
+        assert(ss.rdstate() == ios_base::goodbit);
+        assert(ss.str() == "%Ei%!%E%J%P");
+    }

```

Comment:
This test coverage is great! I'd like to extend it a bit:

* Let's test the fast path for percent escaping. I'm sure we exercise it elsewhere, but covering it here would be good too.
* Let's test that `wchar_t`s aren't truncated, both with and without modifiers.
* `%E%J` here is interesting - it's actually being parsed as `%E%` followed by `J`. Which is fine, but it implies that we should also test something like `%E%Z`. `%Z` is a known conversion specifier, but if we see `%E%` followed by `Z`, both are copied literally.